src/components/Tag/__stories__/index.stories.tsx

```diff
@@ -1,33 +1,109 @@
 import { Meta, Story } from '@storybook/react'
 import { ComponentProps } from 'react'
 import Tag from '..'
+import { SENTIMENTS } from '../../../theme'
 
 export default {
-  args: {
-    children: 'default',
+  component: Tag,
+  title: 'Components/Data Display/Tag',
+} as Meta
+
+const Template: Story<ComponentProps<typeof Tag>> = args => (
+  <Tag {...args}>Tag</Tag>
+)
+
+export const Default = Template.bind({})
+
+export const Variants = Template.bind({})
+Variants.parameters = {
+  docs: {
+    description: {
+      story:
+        'Variants defines different colors of you component. You can define it using `variant` property.',
+    },
   },
-  argTypes: {
-    children: {
-      defaultValue: 'This is a children',
-      name: 'children',
-      table: {
-        defaultValue: { summary: 'This is a children' },
-        type: { required: true, summary: 'string' },
-      },
-      type: { name: 'string', required: true },
+}
+Variants.decorators = [
+  () => (
+    <div style={{ display: 'flex', flexWrap: 'wrap', gap: 8 }}>
+      {SENTIMENTS.map(sentiment => (
+        <Tag key={sentiment} variant={sentiment} onClose={() => {}}>
+          {sentiment}
+        </Tag>
+      ))}
+    </div>
+  ),
+]
+
+export const Icons = Template.bind({})
+Icons.parameters = {
+  docs: {
+    description: {
+      story:
+        'Add Icon on left side of your tag, You can define it using `icon` property.',
     },
   },
-  component: Tag,
-  parameters: {
-    docs: {
-      description: {
-        component: 'Compact element to display information inside a box.',
-      },
+}
+Icons.decorators = [
+  () => (
+    <div style={{ display: 'flex', flexWrap: 'wrap', gap: 8 }}>
+      {SENTIMENTS.map(sentiment => (
+        <Tag key={sentiment} variant={sentiment} icon="check">
+          {sentiment}
+        </Tag>
+      ))}
+    </div>
+  ),
+]
+
+export const onClose = Template.bind({})
+onClose.parameters = {
+  docs: {
+    description: {
+      story:
+        '`onClose` property to pass function trigger on click on close button.',
     },
   },
-  title: 'Components/Data Display/Tag',
-} as Meta
+}
+onClose.decorators = [
+  () => (
+    <div style={{ display: 'flex', flexWrap: 'wrap', gap: 8 }}>
+      {SENTIMENTS.map(sentiment => (
+        <Tag
+          key={sentiment}
+          variant={sentiment}
+          icon="check"
+          onClose={() => {}}
+        >
+          {sentiment}
+        </Tag>
+      ))}
+    </div>
+  ),
+]
 
-const Template: Story<ComponentProps<typeof Tag>> = args => <Tag {...args} />
+export const Loading = Template.bind({})
+Loading.parameters = {
+  docs: {
+    description: {
+      story:
+        'Loading for async operation or other use cases, You can define it by passing `isLoading` property to true.',
+    },
+  },
+}
```

Comment:
For Icons, onClose and Loading story you don't need to display the property for each variant and so you don't need to use `Icons.decorators`. Instead use `Icons.args` it will make story shorter and more clear to understand & read.